src/components/Subthoughts.js

```diff
@@ -395,15 +409,33 @@ export const SubthoughtsComponent = ({
   const editIndex = cursor && children && show ? children.findIndex(child => {
     return cursor[depth] && cursor[depth].rank === child.rank
   }) : 0
-  const filteredChildren = children.filter(child => {
+
+  const _filteredChildren = children.filter(child => {
     const value = showContexts ? head(child.context) : child.value
+    const childContext = pathToContext(unroot(thoughtsRanked)).concat(value)
+    const isPinned = sortPreference === 'Alphabetical' && (hasChild(state, childContext, '=pinnedTop') || hasChild(state, childContext, '=pinnedBottom'))
     return showHiddenThoughts ||
       // exclude meta thoughts when showHiddenThoughts is off
-      (!isFunction(value) && !hasChild(state, unroot(pathToContext(thoughtsRanked).concat(value)), '=hidden')) ||
+      (!isFunction(value) && !hasChild(state, unroot(pathToContext(thoughtsRanked).concat(value)), '=hidden') && !isPinned) ||
       // always include thoughts in cursor
-      (cursor && equalThoughtRanked(cursor[thoughtsRanked.length], child))
+      (cursor && equalThoughtRanked(cursor[thoughtsRanked.length], child) && !isPinned)
   })
 
+  const topPinned = children.filter(child => {
+    const value = showContexts ? head(child.context) : child.value
+    const childContext = pathToContext(unroot(thoughtsRanked)).concat(value)
+    return hasChild(state, childContext, '=pinnedTop')
+  })
+
+  const bottomPinned = children.filter(child => {
+    const value = showContexts ? head(child.context) : child.value
+    const childContext = pathToContext(unroot(thoughtsRanked)).concat(value)
+    return hasChild(state, childContext, '=pinnedBottom')
+  })
```

Comment:
Notice how similar these blocks of code are. Try to abstract the similarities and parameterize the differences to reduce redundancy.